src/main/java/com/redhat/devtools/lsp4ij/features/inlayhint/LSPInlayHintsProvider.java

```diff
@@ -51,8 +52,7 @@ public class LSPInlayHintsProvider extends AbstractLSPDeclarativeInlayHintsProvi
     @Override
     protected void doCollect(@NotNull PsiFile psiFile,
                              @NotNull Editor editor,
-                             @NotNull InlayTreeSink inlayHintsSink,
-                             @NotNull List<CompletableFuture<?>> pendingFutures) {
```

Comment:
Please keep the `@NotNull List<CompletableFuture<?>> pendingFutures)`  parameters. Class which implement doCollect must don't know how pendingFutures is stored.